transaction/std/src/memo/gift_code_sender.rs

```diff
@@ -0,0 +1,190 @@
+        if src.len() > Self::MEMO_DATA_LEN {
+            return Err(MemoError::BadLength(src.len()));
+        }
+
+        let mut memo_data = [0u8; Self::MEMO_DATA_LEN];
+        memo_data[0..src.len()].copy_from_slice(src.as_bytes());
+        Ok(Self { memo_data })
+    }
+}
+
+impl_memo_type_conversions! { GiftCodeSenderMemo }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_gift_code_sender_memo_to_and_from_str() {
+        // Create memo with note
+        let note = "Dear Kitty, you received cash money MeowbleCoin UwU";
+        let memo = GiftCodeSenderMemo::new(note).unwrap();
+
+        // Check that the note is extracted properly
+        assert_eq!(memo.sender_note().unwrap(), note);
+    }
+
+    #[test]
+    fn test_gift_code_sender_memo_with_blank_note_is_ok() {
+        // Create memo with blank note
+        let note = "";
+        let memo = GiftCodeSenderMemo::new(note).unwrap();
+
+        // Check that the note is extracted properly
+        assert_eq!(memo.sender_note().unwrap(), note);
+    }
+
+    #[test]
+    fn test_gift_code_sender_memo_with_only_null_bytes_is_okay() {
+        // Create memo from null bytes
+        let memo_bytes = [0u8; GiftCodeSenderMemo::MEMO_DATA_LEN];
+        let memo = GiftCodeSenderMemo::from(&memo_bytes);
+        let note = "";
+
+        // Check that the note is extracted properly to a blank note
+        assert_eq!(memo.sender_note().unwrap(), note);
+    }
+
+    #[test]
+    fn test_gift_code_sender_note_terminates_at_first_null() {
+        // Create note from bytes with two null bytes
+        let mut note_bytes = [b'6'; 8];
+        note_bytes[3] = 0;
+        note_bytes[6] = 0;
+        let note = "666";
+
+        // Create memo from note bytes
+        let mut memo_bytes = [0u8; GiftCodeSenderMemo::MEMO_DATA_LEN];
+        memo_bytes[0..8].copy_from_slice(&note_bytes);
+        let memo = GiftCodeSenderMemo::from(&memo_bytes);
+
+        // Check that the note is extracted properly and terminated at first null
+        assert_eq!(memo.sender_note().unwrap(), note);
+    }
+
+    #[test]
+    fn test_gift_code_sender_memo_created_with_notes_near_max_byte_lengths() {
+        // Create notes near max length
+        let note_len_minus_one =
+            str::from_utf8(&[b'6'; GiftCodeSenderMemo::MEMO_DATA_LEN - 1]).unwrap();
+        let note_len_exact = str::from_utf8(&[b'6'; GiftCodeSenderMemo::MEMO_DATA_LEN]).unwrap();
+        let note_len_plus_one =
+            str::from_utf8(&[b'6'; GiftCodeSenderMemo::MEMO_DATA_LEN + 1]).unwrap();
+
+        // Create memos from notes
+        let memo_len_minus_one = GiftCodeSenderMemo::new(note_len_minus_one).unwrap();
+        let memo_len_exact = GiftCodeSenderMemo::new(note_len_exact).unwrap();
+        let memo_len_plus_one = GiftCodeSenderMemo::new(note_len_plus_one);
+
+        // Check note lengths match or error on creation if too large
+        let _memo_err: Result<GiftCodeSenderMemo, MemoError> =
```

Comment:
Nit: underscore prefix is for unused variables

Reply:
^^ Same as above, maybe there's a different way?

Reply:
Found an alternate solution